File: app/src/main/java/org/d3ifcool/jagosholat/ui/fragment/JadwalFragment.java
package org.d3ifcool.jagosholat.ui.fragment;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.d3ifcool.jagosholat.util.helper.MethodHelper;
import org.d3ifcool.jagosholat.util.helper.WaktuShalatHelper;
import org.d3ifcool.jagosholat.R;

import static org.d3ifcool.jagosholat.util.helper.VarConstants.Constants;


public class JadwalFragment extends Fragment {

    // ---------------------------------------------------------------------------------------------
    private int countTime;
    // ---------------------------------------------------------------------------------------------

    public JadwalFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.fragment_jadwal, container, false);
        // -----------------------------------------------------------------------------------------
        // Deklarasi Class helper
        WaktuShalatHelper mWaktuShalatHelper = new WaktuShalatHelper();
        MethodHelper mMethodHelper = new MethodHelper();
        // -----------------------------------------------------------------------------------------
        int jumlahDetikSaatIni = mMethodHelper.getSumWaktuDetik();
        String mJadwal = mWaktuShalatHelper.getJadwalShalat();
        // -----------------------------------------------------------------------------------------
        int detikShubuh = mWaktuShalatHelper.getJmlWaktuShubuh();
        int detikDzuhur = mWaktuShalatHelper.getJmlWaktuDzuhur();
        int detikAshar = mWaktuShalatHelper.getJmlWaktuAshar();
        int detikMaghrib = mWaktuShalatHelper.getJmlWaktuMaghrib();
        int detikIsya = mWaktuShalatHelper.getJmlWaktuIsya();
        int detikAfterMid = mWaktuShalatHelper.getJmlAftMidnight();
        int detikBeforeMid = mWaktuShalatHelper.getJmlBeMidnight();
        // -----------------------------------------------------------------------------------------
        // Deklarasi Elemen XML
        TextView mTextViewCoundown = rootView.findViewById(R.id.jadwal_textview_hitungmundur);
        TextView mTextViewWaktuShubuh = rootView.findViewById(R.id.jadwal_textview_shubuh);
        TextView mTextViewWaktuDzuhur = rootView.findViewById(R.id.jadwal_textview_dzuhur);
        TextView mTextViewWaktuAshar = rootView.findViewById(R.id.jadwal_textview_ashar);
        TextView mTextViewWaktuMaghrib = rootView.findViewById(R.id.jadwal_textview_maghrib);
        TextView mTextViewWaktuIsya = rootView.findViewById(R.id.jadwal_textview_isya);
        TextView mTextViewShalatMendatang = rootView.findViewById(R.id.jadwal_textview_shalat);
        // -----------------------------------------------------------------------------------------
        switch (mJadwal) {
            case Constants.SHUBUH:
                mTextViewShalatMendatang.setText(Constants.DZUHUR.substring(7));
                countTime = (detikDzuhur - jumlahDetikSaatIni) * Constants.DETIK_KE_MILI;
                break;
            case Constants.DZUHUR:
                mTextViewShalatMendatang.setText(Constants.ASHAR.substring(7));
                countTime = (detikAshar - jumlahDetikSaatIni) * Constants.DETIK_KE_MILI;
                break;
            case Constants.ASHAR:
                mTextViewShalatMendatang.setText(Constants.MAGHRIB.substring(7));
                countTime = (detikMaghrib - jumlahDetikSaatIni) * Constants.DETIK_KE_MILI;
                break;
            case Constants.MAGHRIB:
                mTextViewShalatMendatang.setText(Constants.ISYA.substring(7));
                countTime = (detikIsya - jumlahDetikSaatIni) * Constants.DETIK_KE_MILI;
                break;
            case Constants.ISYA:
                mTextViewShalatMendatang.setText(Constants.SHUBUH.substring(7));
                if ((jumlahDetikSaatIni == detikAfterMid) || (jumlahDetikSaatIni < detikShubuh)) {
                    countTime = (detikShubuh - jumlahDetikSaatIni) * Constants.DETIK_KE_MILI;
                } else if ((jumlahDetikSaatIni == detikIsya) || (jumlahDetikSaatIni <= detikBeforeMid)) {
                    countTime =  (detikShubuh + detikBeforeMid - jumlahDetikSaatIni) * Constants.DETIK_KE_MILI;
                }
                break;
            default:
                mTextViewShalatMendatang.setText(Constants.DZUHUR.substring(7));
                countTime = (detikDzuhur - jumlahDetikSaatIni) * Constants.DETIK_KE_MILI;
                break;
        }
        // -----------------------------------------------------------------------------------------
        mWaktuShalatHelper.setTimeOnline(mTextViewWaktuShubuh, mTextViewWaktuDzuhur, mTextViewWaktuAshar, mTextViewWaktuMaghrib, mTextViewWaktuIsya);
        mWaktuShalatHelper.CoundownTime(countTime, mTextViewCoundown);
        // -----------------------------------------------------------------------------------------
        return rootView;
    }
}

File: app/src/main/java/com/frogobox/praybox/mvvm/main/JadwalFragment.kt
package com.frogobox.praybox.mvvm.main

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.frogobox.praybox.R
import com.frogobox.praybox.core.BaseFragment
import com.frogobox.praybox.util.helper.ConstHelper
import com.frogobox.praybox.util.helper.MethodHelper
import com.frogobox.praybox.util.helper.WaktuShalatHelper
import kotlinx.android.synthetic.main.fragment_jadwal.*

class JadwalFragment : BaseFragment() {

    private var countTime = 0

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? { // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_jadwal, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // Deklarasi Class helper
        val mWaktuShalatHelper = WaktuShalatHelper()
        val mMethodHelper = MethodHelper()

        val jumlahDetikSaatIni = mMethodHelper.sumWaktuDetik
        val mJadwal = mWaktuShalatHelper.jadwalShalat

        val detikShubuh = mWaktuShalatHelper.jmlWaktuShubuh
        val detikDzuhur = mWaktuShalatHelper.jmlWaktuDzuhur
        val detikAshar = mWaktuShalatHelper.jmlWaktuAshar
        val detikMaghrib = mWaktuShalatHelper.jmlWaktuMaghrib
        val detikIsya = mWaktuShalatHelper.jmlWaktuIsya
        val detikAfterMid = mWaktuShalatHelper.jmlAftMidnight
        val detikBeforeMid = mWaktuShalatHelper.jmlBeMidnight

        when (mJadwal) {
            ConstHelper.Const.SHUBUH -> {
                jadwal_textview_shalat.text = ConstHelper.Const.DZUHUR.substring(7)
                countTime = (detikDzuhur - jumlahDetikSaatIni) * ConstHelper.Const.DETIK_KE_MILI
            }
            ConstHelper.Const.DZUHUR -> {
                jadwal_textview_shalat.text = ConstHelper.Const.ASHAR.substring(7)
                countTime = (detikAshar - jumlahDetikSaatIni) * ConstHelper.Const.DETIK_KE_MILI
            }
            ConstHelper.Const.ASHAR -> {
                jadwal_textview_shalat.text = ConstHelper.Const.MAGHRIB.substring(7)
                countTime = (detikMaghrib - jumlahDetikSaatIni) * ConstHelper.Const.DETIK_KE_MILI
            }
            ConstHelper.Const.MAGHRIB -> {
                jadwal_textview_shalat.text = ConstHelper.Const.ISYA.substring(7)
                countTime = (detikIsya - jumlahDetikSaatIni) * ConstHelper.Const.DETIK_KE_MILI
            }
            ConstHelper.Const.ISYA -> {
                jadwal_textview_shalat.text = ConstHelper.Const.SHUBUH.substring(7)
                if (jumlahDetikSaatIni == detikAfterMid || jumlahDetikSaatIni < detikShubuh) {
                    countTime = (detikShubuh - jumlahDetikSaatIni) * ConstHelper.Const.DETIK_KE_MILI
                } else if (jumlahDetikSaatIni == detikIsya || jumlahDetikSaatIni <= detikBeforeMid) {
                    countTime = (detikShubuh + detikBeforeMid - jumlahDetikSaatIni) * ConstHelper.Const.DETIK_KE_MILI
                }
            }
            else -> {
                jadwal_textview_shalat.text = ConstHelper.Const.DZUHUR.substring(7)
                countTime = (detikDzuhur - jumlahDetikSaatIni) * ConstHelper.Const.DETIK_KE_MILI
            }
        }
        // -----------------------------------------------------------------------------------------
        mWaktuShalatHelper.setTimeOnline(jadwal_textview_shubuh, jadwal_textview_dzuhur, jadwal_textview_ashar, jadwal_textview_maghrib, jadwal_textview_isya)
        mWaktuShalatHelper.CoundownTime(countTime, jadwal_textview_hitungmundur)
        // -----------------------------------------------------------------------------------------
    }
}